src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -1111,6 +1131,16 @@ def assign_role(
                 assign the role. If this is not provided, the role will be
                 assigned globally.
         """
+        path = (
+            f"{USERS}/{str(user_or_team_name_or_id)}{ROLES}"
+            f"?role_name_or_id={role_name_or_id}"
+        )
```

Comment:
You're not checking the `is_user` flag because this obviously only takes care of the user role assignment. You'll also need to address the team role assignment here at some point (by calling the team endpoint).

Reply:
On second thought, you should at least check `is_user` here and raise `NotImplemented` for the team case.